code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -59,8 +59,8 @@
 	if(istype(back,/obj/item/weapon/rig))
 		process_rig(back)
 
-/mob/living/carbon/human/proc/process_glasses(var/obj/item/clothing/glasses/G)
-	if(G && G.active)
+/mob/living/carbon/human/proc/process_glasses(var/obj/item/clothing/glasses/G, var/forceActive)
+	if(G && (G.active || forceActive))
```

Comment:
What's the need for this forceactive flag? To override cell dependancy right?

But even hardsuit vision modules have cell dependancy, the suit has its own internal power source
Checking the active flag should be enough, and the suit should set and control that active flag internally when appropriate

Reply:
Modules have their own power drain, vision modules use an instance of their glasses internally to pass along to vision processors for the proper effects. Glasses keep an internal cell that they use while active in Process(). Neither the glasses cell nor the glasses instance can be accesses by players. The active flag for glasses does not come into play here as it depends on the glasses cell.

Rig could charge/discharge the internal cell of the glasses, or the glasses could check for being inside a module and utilize rig power, but that is complicating things for no gain since module power drain functionality is already in place.

Regardless, balancing is not the scope of this PR, merely restoring lost functionality. A future PR can deal with balancing the module power drain should that be desired.